torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py

```diff
@@ -76,11 +76,38 @@ def forward(self, *args, **kwargs):
             with save_on_cpu(pin_memory=True):
                 return self._checkpoint_wrapped_module(*args, **kwargs)
         else:
-            return self.checkpoint_fn(  # type: ignore[misc]
-                self._checkpoint_wrapped_module,
-                *args,
-                **kwargs
-            )
+            # Support keyword arguments for reentrant checkpoint. Note that this
+            # only works if user has specified self.checkpoint_impl and is not
+            # using their own custom checkpoint_fn.
```

Comment:
Filed https://github.com/pytorch/pytorch/issues/85025 to track this, but the high level is:

1. checkpoint_fn is needed because some use cases (internal) want to use a checkpoint function other than torch.utils.checkpoint with the wrapper, in this case we'll ignore checkpoint_impl and use the passed in function.
2. offload_to_cpu=True is definitely not well designed at the moment, we should revamp.